docs: Add Depot to GitHub integration page (#8768)
dagger/dagger: docs/current_docs/integrations/depot.mdx

---
slug: /integrations/depot
---

# Depot

Depot provides managed, Dagger-powered GitHub Actions runners. These runners, which serve as drop-in replacements for GitHub's own runners, are optimized to make it faster and easier to run Dagger pipelines in GitHub repositories.

## How it works

> When running a CI pipeline on a Dagger-powered Depot GitHub Runner, the general workflow looks like this:

Reviewer: ```suggestion
When running a CI pipeline on a Dagger-powered Depot GitHub Actions Runner, the general workflow looks like this:
```

1. GitHub emits a repository event which triggers a workflow

> 1. Depot receives this event, process it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Runner.

Reviewer: ```suggestion
1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.
```

> 1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.

Author: ```suggestion
1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner. The Depot GitHub Actions Runner has the Dagger CLI pre-installed.
```

> 1. Each Depot GitHub Runner is already connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). Previous runs may have cached data in this external Dagger Engine.

Reviewer: ```suggestion
1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger is pre-configured to cache your previous runs out of the box.
```

> 1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger is pre-configured to cache your previous runs out of the box.

Author: ```suggestion
1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger Engine comes with caching pre-configured.
```

> 1. The "Dagger for GitHub" GitHub Action installs the Dagger CLI, and then calls the specified sub-command, module, function name, and arguments.

Reviewer: The Dagger CLI is pre-installed into our runner.

Author: Fixed

1. The Dagger CLI running inside a Depot GitHub Runner gets automatically connected to Dagger Cloud.

> 1. The Dagger CLI is also configured to connect to the external Dagger Engine which is already

Reviewer: ```suggestion
1. The Dagger CLI is also pre-configured to connect to the external Dagger Engine with persistent caching
```
